cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -149,8 +152,9 @@ handleSimpleNode :: forall blk. RunNode blk
                  -> Tracers ConnectionId blk
                  -> NodeCLI
                  -> NodeConfiguration
+                 -> (NodeKernel IO ConnectionId blk -> IO ())
```

Comment:
@Jimbo4350, two points:

1. I'm hesitant to introduce coupling like this into `handleSimpleNode`, which is currently abstract re: the TUI -- the latter only being introduced at the `runNode` level.
2. I've modelled this callback after `Ouroboros.Consensus.Node.run` -- as in https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/src/Ouroboros/Consensus/Node.hs#L105

How do these arguments factor in your view of things?

Reply:
You have to be careful with a callback like this. It is not allowed to block, or the node will not actually run. This is clearly documented for `Node.run` but is not documented here.

Reply:
@dcoutts, will it be acceptable, if I document that?

Reply:
@dcoutts, added the documentation.

Reply:
@deepfire I was waiting on some feedback from @erikd but I will dismiss my review as its a very small change if we decide what I suggested is favorable. I'm not sure what the benefits are of having this as a callback vs passing in the `MVar` itself (which to me is more readable).

Reply:
Sorry, had a crazy week!

Documenting that something is unsafe is fine if it can't be made unsafe, but it if can, make it safe is always better than documenting that it may be unsafe.